assignment3/Q1.cpp

```diff
@@ -0,0 +1,125 @@
+#include <iostream>
+#include <vector>
+#include <list>
+using namespace std;
+
+
+
+list<string> searchFromPosition(int rows, int cols, char**board, bool **used, vector<string> dict, int x, int y, string word) {
+
+    list<string> part;
+
+    //unavailable position
+    if(board[x][y] == '0' || (used[x-1][y-1] == true))
+        return part;
+
+    word = word + board[x][y];
+    //present string is not prefix
+    if(!dict.isPrefix(word)) {
```

Comment:
`dict` is of type vector, which does not have `isPrefix` and `isWord` functions.
It should be of type `Dictionary` which has these functions.
In order to test your code, you can implement `Dictionary` in any way you want. But currently the code won't even compile.
You don't have to submit your testing code, but your solution should compile and be "ready" to be used and tested.